js/ui/grid_core/ui.grid_core.columns_controller.js

```diff
@@ -1583,9 +1597,12 @@ export const columnsControllerModule = {
                     return result;
                 },
                 getChooserColumns: function(getAllColumns) {
-                    const columns = getAllColumns ? this.getColumns() : this.getInvisibleColumns();
+                    let columns = getAllColumns ? this.getColumns() : this.getInvisibleColumns();
+                    columns = grep(columns, function(column) { return column.showInColumnChooser; });
```

Comment:
Use separate variable for filtered columns. Also, improve this code using filter instead of grep and arrow function for condition:
`const columnChooserColumns = columns.filter(column => column.showInColumnChooser)`